From 19caaac9ad5e92b8c42d036ea6d7886d7849bad9 Mon Sep 17 00:00:00 2001
From: Minecrell <minecrell@minecrell.net>
Date: Sat, 20 Feb 2021 12:25:55 +0100
Subject: [PATCH] net: qualcomm: bam-dmux: Use netif_receive_skb(skb)

bam_dmux_rx_callback() is called in tasklet context, so we should use
netif_rx() instead of netif_rx_ni(). But actually we can apparently
even use netif_receive_skb(skb) in tasklet context. Not sure about
the exact difference... :D
---
 drivers/net/ethernet/qualcomm/bam-dmux.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/qualcomm/bam-dmux.c b/drivers/net/ethernet/qualcomm/bam-dmux.c
index 749ae8d25d7f..6eb094fe689c 100644
--- a/drivers/net/ethernet/qualcomm/bam-dmux.c
+++ b/drivers/net/ethernet/qualcomm/bam-dmux.c
@@ -573,7 +573,7 @@ static bool bam_dmux_cmd_data(struct bam_dmux_skb_dma *skb_dma)
 		break;
 	}
 
-	netif_rx_ni(skb);
+	netif_receive_skb(skb);
 
 	if (bam_dmux_skb_dma_queue_rx(skb_dma, GFP_ATOMIC))
 		dma_async_issue_pending(dmux->rx);
-- 
2.30.1

